meson.build

```diff
@@ -41,6 +41,13 @@ if get_option('readline-provider') == 'readline'
 	if readline.found()
 		add_project_arguments('-DHAVE_READLINE', language: 'c')
 	endif
+
+  rl_replace_line_test = '''#include<readline/readline.h>
+int main() { rl_replace_line; }
+'''
+  if cc.compiles(rl_replace_line_test, name : 'rl_replace_line is defined')
```

Comment:
We can simplify this by using `has_function`.

Reply:
Thanks! I went ahead and did just that. I'm not sure what about the semantics of `has_function` (or arch's read makes the build fail on arch though.